tutorials/scripting/c_sharp/c_sharp_signals.rst

```diff
@@ -0,0 +1,147 @@
+
+.. code-block:: csharp
+
+    [Signal]
+    public delegate void SayHelloEventHandler();
+
+    [Signal]
+    public delegate void SayHelloWithNameEventHandler(string myString);
+
+Once this is done, Godot will create the appropriate events automatically behind the scenes. You
+can then use said events as you'd do for any other Godot signal. Note that events are named using
+your delegate's name minus the final ``EventHandler`` part.
+
+.. code-block:: csharp
+
+    public override void _Ready()
+    {
+        SayHello += () => GD.Print("Hello!");
+        SayHelloWithName += SayHelloTo;
+    }
+
+    private void SayHelloTo(string name)
+    {
+        GD.Print($"Hello {name}!");
+    }
+
+.. warning::
+
+    If you want to connect to these signals in the editor, you will need to (re)build the project
+    to see them appear.
+
+    You can click the **Build** button in the upper-right corner of the editor to do so.
+
+Signal emission
+---------------
+
+To emit signals, use the ``EmitSignal`` method. Note that, as for signals defined by the engine,
+your custom signal names are listed under the nested ``SignalName`` class.
+
+.. code-block:: csharp
+
+    public void MyMethodEmittingSignals()
+    {
+        EmitSignal(SignalName.SayHello);
+        EmitSignal(SignalName.SayHelloWithName, "World");
+    }
+
+Signals support arguments of:
+
+* All the `built-in types <https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/built-in-types-table>`_,
+* Classes derived from :ref:`Godot.Object <class_Object>`
+* Collections of those types
+
+Consequently, any ``Node`` or ``Reference`` will be compatible automatically, but custom data objects will need
+to inherit from ``Godot.Object`` or one of its subclasses.
+
+.. code-block:: csharp
+
+    public partial class DataObject : Godot.Object
+    {
+        public string MyFirstString { get; set; }
+        public string MySecondString { get; set; }
+    }
+
+Bound values
+------------
+
+Sometimes you'll want to bind values to a signal when the connection is established, rather than
+(or in addition to) when the signal is emitted. To do so, you can use an anonymous function like in
+the following example.
+
+Here, the :ref:`Button.Pressed <class_BaseButton_signal_pressed>` signal do not take any argument. But we
+want to use the same ``ModifyValue`` for both the "plus" and "minus" buttons. So we bind the
+modifier value at the time we're connecting the signals.
+
+.. code-block:: csharp
+
+    public int Value { get; private set; } = 1;
+
+    public override void _Ready()
+    {
+        Button plusButton = GetNode<Button>("PlusButton");
+        plusButton.Pressed += () => ModifyValue(1);
+
+        Button minusButton = GetNode<Button>("MinusButton");
+        minusButton.Pressed += () => ModifyValue(-1);
+    }
+
+    private void ModifyValue(int modifier)
+    {
+        Value += modifier;
+    }
+
+Signal creation at runtime
```

Comment:
It's probably not necessary but we may want to add a note that adding signals this way won't allow you to use them as events or use `SignalName`.